content/articles/lxc-vs-docker-what-is-the-difference-and-why-docker-is-better/index.md

```diff
@@ -0,0 +1,109 @@
+### What is LXC, and what does it do
+
+[LXC](https://linuxcontainers.org/LXC/introduction/) is a userspace interface for the [Linux kernel containment features](https://linuxcontainers.org/LXC/getting-started/). It uses a sophisticated API and simple tools to make it easy for Linux users to develop and maintain systems or application containers.
+
+To understand LXC well, let first discuss what a virtual machine is.
+
+#### The concept of Virtual Machines (VM)
+
+Typically, in a virtual machine, you have a host machine where you install your operating system. It could be Windows, macOS, or Linux on the host machine. When you use a virtual machine, you add or install hypervisors in your host operating system, for example, VirtualBox, VMware, and ESXi. You use the hypervisor to install a guest operating system to run on top of your host operating system. In this case, the guest OS is running as a virtual machine/container. Each of the guest operating systems you add is a complete operating system. all the hardware resources (RAM, CPU, HDD/SSD) attached to these guest virtual machines are virtualized. And the hypervisor is responsible for allocating the guest OS the hardware resources they need from your base/host machine in a virtualized fashion. Basically, you create an instance of an emulator that you can interact with just like an ordinary computer. The image below depicts this concept.
+
+![Virtual machine architecture](/engineering-education/lxc-vs-docker-what-is-the-difference-and-why-docker-is-better/virtual-machine-VM-architecture.jpeg)
+
+[Image source](https://www.docker.com/resources/what-container)
+
+The main issue with the above is that every instance running will need resource allocation, which can lead to over-allocation of the available resources. Every instance of an operating system has to have these resources allocated to it, and those resources will be used whether or not the server needs them.
+
+Check this out. If you have 16GB of RAM and start 4 Guest machines/servers, they will use all of the RAM that is currently available, i.e., each is allocated 4GB of ram. Even if the individual server only uses 1GB of RAM. Assuming each can run on 1GB of RAM, you just need 4RAM to boot up these four virtual hosts. But because the instances that they are in have been given four gigs each, they are using 16GB of ram within your virtual environment. If you were to add another virtual host, you wouldn't have enough RAM to start it. Every time a VM starts, all of the resources that have been assigned to it are automatically given to it. Those resources can't be assigned to another guest host. In the end, you have lots of wasted resources.
+
+Another issue is that all the guest operating instances are running on a single machine. In case of any boot-up processes, or whenever you want to restart service for some reason, you have to reboot that entire computer. depending on what environment you are you're in, a couple of extra seconds matter
```

Comment:
```suggestion
Another issue is that all the guest operating instances are running on a single machine. In case of any boot-up processes, or whenever you want to restart the service for some reason, you have to reboot the entire computer. Depending on what environment you are you're in, a couple of extra seconds matter.
```